bot/resources/tags/type-hint.md

```diff
@@ -0,0 +1,16 @@
+**Type Hints**
+
+A typehint indicates what type something should be. For example,
+```python
+def add(a: int, b: int) -> int:
+    sum: int = a + b
+    return sum
+```
+In this case, `a` and `b` are expected to be ints, and the function returns an int. We also declare an intermediate variable `sum`, which we indicate to be an int.
+
+It's important to note these are just hints and have no runtime effect. For example,
+```python
+#uh oh
+add("hello ", "world")
+```
+This code will run without error, even though it doesn't follow the function's type hints.
```

Comment:
```suggestion
This code will run without error because strings can also be added together, even though it doesn't follow the function's type hints.
```

Something like this or similar, we should make it clear that it works even though it's the wrong type because the type we passed in allows the same operations - a kind of gateway into duck typing.